commit f7e3fb0a6f9d6bf3b8bf33f6205c2548f426ce88
Author: Felix Fietkau <nbd@nbd.name>
Date:   Fri Jan 25 08:21:26 2019 +0100

    batman-adv: release station info tidstats
    
    [ Upstream commit 7d652669b61d702c6e62a39579d17f6881670ab6 ]
    
    With the addition of TXQ stats in the per-tid statistics the struct
    station_info grew significantly. This resulted in stack size warnings
    due to the structure itself being above the limit for the warnings.
    
    To work around this, the TID array was allocated dynamically. Also a
    function to free this content was introduced with commit 7ea3e110f2f8
    ("cfg80211: release station info tidstats where needed") but the necessary
    changes were not provided for batman-adv's B.A.T.M.A.N. V implementation.
    
    Signed-off-by: Felix Fietkau <nbd@nbd.name>
    Fixes: 8689c051a201 ("cfg80211: dynamically allocate per-tid stats for station info")
    [sven@narfation.org: add commit message]
    Signed-off-by: Sven Eckelmann <sven@narfation.org>
    Signed-off-by: Simon Wunderlich <sw@simonwunderlich.de>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/net/batman-adv/bat_v_elp.c b/net/batman-adv/bat_v_elp.c
index e8090f099eb8..ef0dec20c7d8 100644
--- a/net/batman-adv/bat_v_elp.c
+++ b/net/batman-adv/bat_v_elp.c
@@ -104,6 +104,9 @@ static u32 batadv_v_elp_get_throughput(struct batadv_hardif_neigh_node *neigh)
 
 		ret = cfg80211_get_station(real_netdev, neigh->addr, &sinfo);
 
+		/* free the TID stats immediately */
+		cfg80211_sinfo_release_content(&sinfo);
+
 		dev_put(real_netdev);
 		if (ret == -ENOENT) {
 			/* Node is not associated anymore! It would be
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1078_f7e3fb0a6f9d6bf3b8bf33f6205c2548f426ce88.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1078_f7e3fb0a6f9d6bf3b8bf33f6205c2548f426ce88.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
